src/hstr_history.c

```diff
@@ -99,11 +99,11 @@ bool is_hist_timestamp(const char* line)
     return (i >= 11);
 }
 
-char* parse_history_line(char *l)
+char* parse_history_line(char *l, bool isZsh)
```

Comment:
Maybe a static variable would be less intrusive and more respectful of the separation of concerns in this function? Currently it is supposed to determine whether a line is from a zsh history or not, but changing the signature as such makes the code spill over.

Something like this maybe:
```diff
 char* parse_history_line(char *l)
 {
+#ifndef HSTR_TESTS_UNIT
+    static bool isZsh, c=0;
+    if(!c) {
+        isZsh = isZshParentShell();
+        c = 1;
+    }
+#endif
+
     if(
 #ifndef HSTR_TESTS_UNIT
-    !isZshParentShell() ||
+    !isZsh ||
 #endif
     !l ||
     l[0]!=':') {
```

Reply:
The suggested changes have been applied.